src/simplewallet/simplewallet.cpp

```diff
@@ -4844,6 +5105,37 @@ bool simple_wallet::stop_mining(const std::vector<std::string>& args)
   return true;
 }
 //----------------------------------------------------------------------------------------------------
+bool simple_wallet::check_daemon_rpc_prices(const std::string &daemon_url, uint32_t &actual_cph, uint32_t &claimed_cph)
+{
+  try
+  {
+    const auto nodes = m_wallet->get_public_nodes(false);
```

Comment:
This compares the price claimed on the P2P list with the price given when trying to mine. The idea is to not give an incentive for nodes to lie on the P2P list since it'd be shown in the wallet when a user actually tries to mine.

Reply:
I get your intention, but I don't see this working. `get_public_nodes` RPC is called against the new daemon that the wallet connects to via the `set_daemon` command. In `core_rpc_server::on_get_public_nodes`, I see that only the p2p peer nodes are included in the response, not the connected node itself. Thus I expect the line 5122 `if (host == daemon_url)` to always evaluate to `false` and thus `check_daemon_rpc_prices` to always return `false`. Or am I missing something?

Reply:
I see the problem. I had this patch to allow testing llocally:

```
+      : id(id), host(epee::string_tools::get_ip_string_from_int32(ip)), ip(ip), port(port), rpc_port(rpc_port), rpc_credits_per_hash(rpc_credits_per_hash), last_seen(ladiff --git a/src/p2p/net_peerlist.h b/src/p2p/net_peerlist.h
index f20cb7b40..1ff057c12 100644
--- a/src/p2p/net_peerlist.h
+++ b/src/p2p/net_peerlist.h
@@ -253,11 +253,11 @@ namespace nodetool
   bool peerlist_manager::is_host_allowed(const epee::net_utils::network_address &address)
   {
     //never allow loopback ip
-    if(address.is_loopback())
-      return false;
+//    if(address.is_loopback())
+//      return false;
 
-    if(!m_allow_local_ip && address.is_local())
-      return false;
+//    if(!m_allow_local_ip && address.is_local())
+//      return false;
 
     return true;
   }
```

I'll change get_public_nodes to include the local node if it provides public RPC.,

Reply:
Except a node doesn't usually know its IP address. I'll rework that.

Reply:
Done, the wallet now remembers the claimed cost from earlier public_node commands, and compares with this.